### PR TITLE
Devdocs: Cleaning up the place

### PR DESCRIPTION
**Pull request by michaeldcain**
This PR updates the layout of `devdocs` to be a little nicer, more consistent, and hopefully cleans things up for more improvements.

Here's some screenshots, the updates on the left, and current wpcalypso on the right:

group

* We do some strange things loading in the SCSS for devdocs. This PR removes references to devdoc SCSS from the main `_components.scss` file, in favor of loading them using the `sections/devdocs.scss` file.
* At larger screen widths, the list of component and block examples will break out into two columns. We toyed with fixed height examples on the list (with an overflow: hidden or scroll) but decided to keep experimenting in another PR coming soon.
* When viewing a single component or block the README.md file is now displayed by default; No more `FoldableCard` component.
* Added a (hopefully) consistent max-width across all screens within devdocs.

### CODE REVIEW

**Comment by drw158** [comment]
I like where you ended up on the design for the UI Components section.

Just some weirdness here:

screen shot 2018-03-15 at 9 26 39 am
screen shot 2018-03-15 at 9 24 10 am
screen shot 2018-03-15 at 9 23 49 am
screen shot 2018-03-15 at 9 29 12 am

While searching, the "Showing" text looks weird. Maybe the text should wrap?

**Comment by michaeldcain** [comment]
@folletto Thanks for the review!

> I'd however consider keeping somewhere "Improve this document on GitHub", as these edit flows help keeping the documents up to date.

It looks like these were moved to the bottom of each doc. I'm fine with moving them back to the top if you think that'd be better.

**Comment by folletto** [comment]
I feel it would be idea if it was very prominent and easy to access, yes. I'm open to suggestions to be clear, and it can ship with it at the bottom too. I just want to nudge people as much as possible that editing is quick ;)

File: assets/stylesheets/sections/devdocs.scss [hunk 1]
```diff
@@ -1,13 +1,4 @@
-// Main stylesheet
```
Comment (gwwar) [inline assets/stylesheets/sections/devdocs.scss]: Did we not need this comment?

File: client/devdocs/style.scss [hunk 2]
```diff
@@ -1,91 +1,489 @@
...
+}
+
+
+// Some components/blocks use a button to toggle variants
+// This is really dumb right now.
+.docs__design-toggle,
+.design__toggle {
+	top: 10px;
+	right: 24px;
+	z-index: 1;
```
Comment (gwwar) [inline client/devdocs/style.scss]: Why do we need z-index for a toggle? Can we not use the natural stacking order?

File: client/devdocs/style.scss [hunk 3]
```diff
@@ -1,91 +1,489 @@
...
+	font-family: $sans;
+	text-align: center;
+	color: $gray-text-min;
+}
+
+
+// Some components/blocks use a button to toggle variants
+// This is really dumb right now.
+.docs__design-toggle,
+.design__toggle {
```
Comment (gwwar) [inline client/devdocs/style.scss]: I think we can remove `design__toggle` I can't find usages in code. Only exception would be folks are constructing class names like `'design_'+name`

File: client/devdocs/style.scss (hunk 2 above)
Comment (michaeldcain) [reply client/devdocs/style.scss]: Without the z-index, it falls behind the header when the column layout is being used.

**Comment by gwwar** [comment]
First off, good looking changes! ✨ Thanks for working on this! 

Since this PR is really big, let's maybe try to wrap up the really obvious bugs, and make other enhancements/fixes in smaller follow up PRs.

The biggest weird one for me is our toggle buttons like so:
screen shot 2018-03-15 at 1 44 49 pm

It is very handy to toggle this on the header, but that's not where this ~~button~~ link lives. We're fighting the natural stacking order, and end up with more CSS, and corner cases. Let's try to move this if possible (with dev help in follow up PRs to the header). **For now can we let the ~~button~~ link live in the top right of the content card?**

screen shot 2018-03-15 at 1 38 16 pm

cc @Automattic/flowpatrol do you mind helping spot any oddities here too?

**Comment by nylen** [comment]
This is awesome, because when this section is working well, it can really help development and design go more quickly.  Thanks for giving it some love :sparkles: 

There is this empty space where the page search/header isn't taking up the full width.  This also looks different than the screenshots in the OP.  Is it a regression, or is it the intended behavior of this PR?

> 

> 

Either way I think this is good to ship, this PR is a big improvement.

**Pull request merged**